rdkit/Chem/SaltRemover.py

```diff
@@ -69,22 +111,29 @@ def _initPatterns(self):
 
     >>> RDLogger.EnableLog('rdApp.error')
     """
-    whitespace = re.compile(r'[\t ]+')
     if self.defnData:
       from rdkit.six.moves import cStringIO as StringIO
       inF = StringIO(self.defnData)
+      with closing(inF):
+        self.salts = []
+        for line in inF:
+          if line:
+            if self.defnFormat == InputFormat.SMARTS:
```

Comment:
this `if` block should have a final `else:` that raises `ValueError('Unsupported format for supplier.')`